### PR TITLE
Support `babel-ts` AST type `TSNamedTupleMember`

### LINKED ISSUE #8883: new AST type `TSNamedTupleMember` not handled
NOTE: this is todo for #8869

**Prettier pr-8869**
[Playground link](https://deploy-preview-8869--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEMCeAHOACAKtgXmwG0APJbAQQBps0B+CgIVoDp2AvCgYQF0BuENRAQMMAJbQAzslABDAE4KIAdwAKihDJRyANirloZwgEYK5YANZwYAZQwXxUAObIYCgK5xhcALYm4ABNAoIAZORcPOWc4ADEIBV85GAkXZBA5DxgIIRAACxhfXQB1PPF4KQcwOFstcvEAN3K0dLApYxAnKTgFGDVzZyTkADM9buEAKylSJnMrG1s5XzhQpzgRse8QKdJbJ2ddOABFDwh4Dd1xkAcFboV0kzkA3QBaGA6MBScYYvFAmDyyAAHAAGYSfCDdYrmDDpT5wO4NdbCACOp3g-VE2gyUheUDgQSCuQUcDR4hJ-WiQyQo0uW26vnEbk89P2hxOZ3WNM2whgT1+-0BSAATLzzOJdPtuBBfNSQAiAKy5DzdXBPbS0q4NLwASSgIVgtjAXzElH1tnQhwu3QAvjagA)
```sh
--parser babel-ts
```

**Input:**
```tsx
type T = [x: A, y?: B, ...z: C];
```

**Output:**
```tsx
Error: unknown type: "TSNamedTupleMember"
    at https://deploy-preview-8869--prettier.netlify.app/lib/standalone.js:15:261132
    at Object.print (https://deploy-preview-8869--prettier.netlify.app/lib/standalone.js:15:261184)
    at ri (https://deploy-preview-8869--prettier.netlify.app/lib/standalone.js:15:107571)
    at https://deploy-preview-8869--prettier.netlify.app/lib/standalone.js:15:107033
    at Object.printComments (https://deploy-preview-8869--prettier.netlify.app/lib/standalone.js:15:103467)
    at e (https://deploy-preview-8869--prettier.netlify.app/lib/standalone.js:15:107014)
    at https://deploy-preview-8869--prettier.netlify.app/lib/standalone.js:15:216013
    at Ga.each (https://deploy-preview-8869--prettier.netlify.app/lib/standalone.js:15:105537)
    at ah (https://deploy-preview-8869--prettier.netlify.app/lib/standalone.js:15:215980)
    at https://deploy-preview-8869--prettier.netlify.app/lib/standalone.js:15:246260
```

**Expected behavior:**

### PR DESCRIPTION
**Pull request by fisker**
Fixes #8883

Include changes from #8869, should merge it first.

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: tests/misc/typescript-babel-only/tuple-labeled.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// https://github.com/babel/babel/pull/11754
+// TODO: move this file to `test/typescript/tuple`
```
Comment (brodybits) [inline tests/misc/typescript-babel-only/tuple-labeled.ts:2]: ??

Comment (sosukesuzuki) [reply tests/misc/typescript-babel-only/tuple-labeled.ts:2]: Current typescript-estree cannot parse labeled tuple types. So we should put this in `/tests/misc/typescript-babel-only` for now.

**Comment by thorn0** [comment]
@Arrow7000 
[Please include a repro](https://gist.github.com/Rich-Harris/88c5fc2ac6dc941b22e7996af05d70ff).

**Comment by Arrow7000** [comment]
Hi @thorn0, thanks for looking into this. I managed to reproduce this on a [CodeSandbox](https://codesandbox.io/s/vibrant-monad-0w56y?file=/src/index.ts). You can invoke Prettier by saving the file.

Let me know if you need anything else from me.

**Comment by thorn0** [comment]
Sorry, but this isn't a valid isolated reproduce. If you can't provide a reproduce that doesn't involve third-party things like VS Code or CodeSandbox, that means that the problem is not on Prettier's side.

**Pull request merged**